src/utils/index.ts

```diff
@@ -0,0 +1,311 @@
+/**
```

Comment:
Please search for each function and make sure they are **used** in the project. Thanks!